lib/Dialect/EmitC/Conversion/TosaToEmitC.cpp

```diff
@@ -223,6 +223,46 @@ class MatMulOpConversion : public OpConversionPattern<tosa::MatMulOp> {
   }
 };
 
+class ReluNOpConversion : public OpConversionPattern<tosa::ReluNOp> {
+  using OpConversionPattern<tosa::ReluNOp>::OpConversionPattern;
+
+public:
+  ReluNOpConversion(MLIRContext *ctx)
+      : OpConversionPattern<tosa::ReluNOp>(ctx) {}
+
+private:
+  LogicalResult
+  matchAndRewrite(tosa::ReluNOp reluNOp, ArrayRef<Value> operands,
+                  ConversionPatternRewriter &rewriter) const override {
+
+    StringRef funcName = "tosa::reluN";
+    StringAttr callee = rewriter.getStringAttr(funcName);
+
+    SmallVector<Attribute, 2> args_;
+    args_.push_back(rewriter.getIndexAttr(0));
+
+    // Since tosa.reluN has two max attribute types for float and integer
+    // values, we have to determine to which max attribute we have to clamp to
+    auto elementType =
+        operands[0].getType().cast<RankedTensorType>().getElementType();
```

Comment:
You may need to check if this is a ranked Tensor. I see two options:

```c++
    if (auto tensorType = operands[0].getType().dyn_cast<RankedTensorType>())
      auto elementType = tensorType.getElementType();
      
      ...
      
    } else {
      return failure();
    }
```

but I would probably go for a more general check. Instead one could go for

```suggestion
    auto tensorType = operands[0].getType().dyn_cast<TensorType>()
    if (!tensorType.hasRank())
      return failure();
    auto elementType = tensorType.getElementType();
```

Reply:
We won't check this here as discussed internally.